### PR TITLE
substantial mod59 struct changes

### PR DESCRIPTION
**Pull request by red031000**
adds message stuff to the mod59 struct, need to be merged so I can use it in CE to continue decompiling

### CODE REVIEW

File: arm9/modules/59/src/mod59_021D74E0_src.c [hunk 1]
```diff
@@ -277,3 +280,44 @@ THUMB_FUNC void MOD59_SetupBg(MOD59_OverlayData *data)
...
+    FreeBgTilemapBuffer(data->bgConfig, GF_BG_LYR_SUB_1);
+    FreeBgTilemapBuffer(data->bgConfig, GF_BG_LYR_SUB_2);
+    FreeBgTilemapBuffer(data->bgConfig, GF_BG_LYR_SUB_3);
+
+    FreeToHeap(data->bgConfig);
+}
+
+THUMB_FUNC void MOD59_SetupMsg(MOD59_OverlayData *data)
+{
+    data->msgData = NewMsgDataFromNarc(1, NARC_MSGDATA_MSG, 0x155, data->heap_id);
```
Comment (tgsm) [inline arm9/modules/59/src/mod59_021D74E0_src.c:311]:
```suggestion
    data->msgData = NewMsgDataFromNarc(1, NARC_MSGDATA_MSG, NARC_msg_narc_0341_bin, data->heap_id);
```
you'll need to `#include "msgdata/msg.naix"`

File: arm9/modules/59/src/mod59_021D74E0_src.c [hunk 2]
```diff
@@ -277,3 +280,44 @@ THUMB_FUNC void MOD59_SetupBg(MOD59_OverlayData *data)
...
+
+    FUN_0201BD5C();
+
+    data->Unk60 = FUN_020142EC(0, 0, 6, data->heap_id);
```
Comment (PikalaxALT) [inline arm9/modules/59/src/mod59_021D74E0_src.c:315]: `unkXX`, `unk_XX`, or `field_XX` are what's used elsewhere in the repo

**Comment by red031000** [comment]
addressed

File: arm9/modules/59/include/mod59_021D74E0.h [hunk 3]
```diff
@@ -20,22 +20,31 @@ typedef struct MOD59_OverlayData
...
-    u8 padding2[0xC];
-    u8 Unk88;
-    u8 Unk89;
-    u8 Unk8A;
-    u8 Unk8B;
-    u8 padding3[4];
-    u32 Unk90;
-    u8 padding4[32];
+    u32 unk78;
+    u8 padding5[0xC];
```
Comment (PikalaxALT) [inline arm9/modules/59/include/mod59_021D74E0.h:40]: Typically, we use `filler_OFFSET` for intermediate unallocated space

**Review by PikalaxALT — APPROVED** [review]

**Pull request merged**